src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3626,16 +3631,261 @@ bool isInstanceConstructor(out MethodSymbol constructorSymbol)
             }
         }
 
-        internal virtual BoundExpressionStatement BindConstructorInitializer(ConstructorInitializerSyntax initializer, BindingDiagnosticBag diagnostics)
+#nullable enable
+        [return: NotNullIfNotNull("initializer")]
+        internal virtual BoundExpressionStatement? BindConstructorInitializer(ConstructorInitializerSyntax? initializer, SyntaxNode? ctorSyntax, BindingDiagnosticBag diagnostics)
```

Comment:
>internal virtual BoundExpressionStatement? BindConstructorInitializer(ConstructorInitializerSyntax? initializer, SyntaxNode? ctorSyntax, BindingDiagnosticBag diagnostics) [](http://example.com/codeflow?start=8&length=170)

Since we are not planning to cache implicit initializer for the purpose of SemanticModel (and it probably doesn't make sense to do because it doesn't have corresponding syntax node and cannot be bound by SemanticModel on its own), I do not think there is any benefit in bundling explicit and implicit case in one method. I recommend leaving original method unchanged and adding dedicated method for implicit initializer (with the name clearly reflecting that). As an additional benefit, majority of changes in SemanticModel will go away.